File: Backend/CatalogoComercial/Models/Produto.cs
using System.ComponentModel.DataAnnotations;

namespace CatalogoComercial.Api.Models;

public class Produto
{
    public int Id { get; set; }

    [Required]
    [MinLength(5)]
    public string Nome { get; set; } = string.Empty;

    public string? Descricao { get; set; }

    public decimal Preco { get; set; }

    public int CategoriaId { get; set; }

    public Categoria? Categoria { get; set; }
}

File: Frontend/types/produto.ts
import type { Categoria } from "./categoria";

export interface Produto {
    id: number
    nome: string
    descricao: string
    preco: number
    categoriaId: number
    categoria: Categoria
  }
  
  export interface ProdutoRequest {
    nome: string
    descricao: string
    preco: number
    categoriaId: number
  }